lib/Serialization/DeserializeSIL.cpp

```diff
@@ -394,6 +394,41 @@ SILDeserializer::readNextRecord(SmallVectorImpl<uint64_t> &scratch) {
   return maybeKind;
 }
 
+std::optional<SILLocation> SILDeserializer::readLoc(unsigned kind,
+                                     SmallVectorImpl<uint64_t> &scratch) {
+  unsigned LocationKind = 0;
+  SILLocation::FilenameAndLocation *FNameLoc = nullptr;
+  if (kind == SIL_SOURCE_LOC_REF) {
```

Comment:
Can you add a short comment explaining the strategy of serializing a sourceloc only the first time it appears and referencing this first sourceloc for successive appearances.